Lib/ast.py

```diff
@@ -1699,6 +1746,23 @@ def unparse(ast_obj):
     return unparser.visit(ast_obj)
 
 
+_deprecated_globals = {
+    name: (globals().pop(name), '; use ast.Constant instead')
+    for name in ('Num', 'Str', 'Bytes', 'NameConstant', 'Ellipsis')
+}
+
+def __getattr__(name):
+    if name in _deprecated_globals:
+        value, details = _deprecated_globals[name]
+        globals()[name] = value
```

Comment:
Q: do we re-insert the value into `globals()` to avoid emiting multiple deprecation warnings, or for performance, or something else?

Reply:
Probably a mixture of the two, but I'm not sure (I'm AFK right now). I took over this PR from Serhiy's https://github.com/python/cpython/pull/31432, and this bit of code looked correct, so I didn't change it!